Add kubestash docs (#1)
kubestash/docs: docs/guides/cli/kubectl-plugin/index.md

| [resume backup](#resume-backup)             | Resume KubeStash backup.                                                   |
| [password add](#password-add)               | Add a new password (key) to restic repositories.                           |
| [password list](#password-list)             | List the passwords (keys) for restic repositories.                         |
| [password remove](#password-remove)         | Remove passwords (keys) from restic repositories                           |
| [password update](#password-update)         | Update current password (key) for restic repositories                      |
| [debug backup](#debug-backup)               | Debug KubeStash backup issues.                                             |
| [debug restore](#debug-restore)             | Debug KubeStash restore issues.                                            |
| [debug operator](#debug-operator)           | Debug KubeStash operator issues.                                           |

## Copy Command

`kubectl kubestash cp` command is used to copy objects like, `Secret` and `VolumeSnapshot`, from one namespace to another namespace.

### Copy Secret

> To copy a Secret, you need to provide `Secret` name and destination namespace. You will provide the destination namespace by using flag. The available flags are:

Reviewer: ```suggestion
To copy a Secret, you need to provide `Secret` name and destination namespace. The available flags are:
```

| Flag             | Description                                                            |
|------------------|------------------------------------------------------------------------|
| `--namespace`    | Indicates the namespace of the respective `Secret`.                    |
| `--to-namespace` | Indicates the destination namespace where the `Secret` will be copied. |

**Format:**

```bash
kubectl kubestash cp secret <secret-name> [flags]
```

**Example:**

```bash
$ kubectl kubestash cp secret my-secret --namespace=demo --to-namespace=demo1
```

### Copy VolumeSnapshot

> To copy a VolumeSnapshot, you need to provide `VolumeSnapshot` name and destination namespace. You will provide the destination namespace by using flag. The available flags are:

Reviewer: ```suggestion
To copy a VolumeSnapshot, you need to provide `VolumeSnapshot` name and destination namespace. The available flags are:
```

| `--to-namespace` | Indicates the destination namespace where the `VolumeSnapshot` will be copied. |

**Example:**

```bash
$ kubectl kubestash cp volumesnapshot my-vol-snap --namespace=demo --to-namespace=demo1
```

## Clone PVC

`kubectl kubestash clone pvc` command is used to clone PVC from one namespace to another namespace. When we run the command the cloning process consists of the following steps:

- At first, It creates a BackupStorage in the source namespace (if no storage is provided).
- It creates a RetentionPolicy in the source namespace.
- Using these BackupStorage and RetentionPolicy, it creates a BackupConfiguration targeting the PVC to take backup.
- After the backup process succeeded, It creates a RestoreSession in the source namespace with the target in destination namespace.
- finally, It restores the backed up data into VolumeClaimTemplate in the destination namespace.

To clone a PVC, you need to provide backend credentials for creating BackupStorage (if no storage is provided) and encryption secret.
You will provide the backend credential and encryption secret by using flags. The available flags are:

> 

Reviewer: To clone a PVC, you can provide the reference of the existing BackupStorage if you already have one; otherwise, you have to provide the backend information. Additionally, you need to provide the encryption secret reference. The available flags are:

Reviewer: I think we can skip taking the encryption secret from the user.

| `--encrypt-secret`           | Specify the name of the encryption secret                             |
| `--encrypt-secret-namespace` | Specify the namespace of the encryption secret.                       |
| `--prefix`                   | Specify the directory inside the backend.                             |
| `--storage-name`             | Specify the name of the BackupStorage.                                |
| `--storage-namespace`        | Specify the namespace of the BackupStorage.                           |

If you want to use existing BackupStorage, then provide `storage-name` and `storage-namespace` flags and there is no need to provide the `provider`, `bucket`, `endpoint`, `region`, `max-connections`, `storage-secret` and `prefix` flags.

**Format:**

```bash
kubectl kubestash clone pvc <pvc-name> [flags]
```

**Example:**

```bash
$ kubectl kubestash clone pvc my-pvc -n demo --to-namespace=demo-1 --storage-secret=<storage-secret> --bucket=<bucket> --prefix=<prefix> --provider=<provider> --encrypt-secret=<encryption-secret> --encrypt-secret-namespace=<encryption-secret-namespace>
```

## Download Snapshot

> `kubectl kubestash download` command is used to download the components of a snapshot from backend repository into your local machine.
> To download the snapshot you have to provide `Snapshot` name and download directory. You will provide the download directory using flag. The available flags are:

Reviewer: ```suggestion
`kubectl kubestash download` command is used to download the components of a snapshot from backend repository into your local machine.
To download the snapshot you have to provide `Snapshot` name and download directory. The available flags are:
```

| `--components`  | Specifies the list of components to download. (If this flag is not provided, then all the components of the snapshot will be downloaded) |
| `--exclude`     | Specifies the list of pattern for directory/file to ignore during download.                                                              |
| `--include`     | Specifies the list of pattern for directory/file to download.                                                                            |

**Format:**

```bash
kubectl kubestash download <snapshot-name> [flags]
```

**Example:**

```bash
$ kubectl kubestash download demo-storage-sample-backup-frequent-backup-1699355040 --namespace=demo --destination=/home/user/downloads/ 
```

## Trigger an Instant Backup

`kubectl kubestash trigger` command is used to take an instant backup in KubeStash.
To trigger an instant backup, you need to have a BackupConfiguration in your cluster. You need to provide the BackupConfiguration name. You can also provide the namespace by using the `--namespace` flag. This flag indicates the namespace where the trigger backup will be occurred. 

> To trigger backup for specific sessions, you need to provide the sessions name (comma seperated) by using the `--sessions` flag. If you want to trigger backup for all sessions present in the BackupConfiguration, then this flag can be omitted.

Reviewer: ```suggestion
To trigger backup for specific sessions, you need to provide the session names (comma seperated) by using the `--sessions` flag. If you want to trigger backup for all the sessions present in the BackupConfiguration, then this flag should be omitted.
```

**Format:**

```bash
$ kubectl kubestash trigger <backupconfig-name> [flags]
```

**Example:**

```bash
$ kubectl kubestash trigger my-config --namespace=demo --sessions=frequent-backup,everyday-backup
```

## Unlock Repository

> 
> `kubectl kubestash unlock` are used to remove lock from the backend repository.
> To unlock the Repository, you need to provide a Repository name. You can also provide the namespace by using the `--namespace` flag. This flag indicates the Repository namespace.
> To unlock the Repository for a specific component, you need to provide the component paths (comma seperated) by using the `--paths` flag. You can find the paths in the `status` section of the respective Repository. If you want to unlock all the components of the respective Repository, then this flag can be omitted.
> 

Reviewer: ```suggestion

`kubectl kubestash unlock` is used to remove lock from the backend repository.

To unlock a Repository, you need to provide a Repository name. You can also provide the namespace by using the `--namespace` flag. This flag indicates the Repository namespace.

A Repository can have multiple components. To unlock the repository for a specific component, you need to provide the component paths (comma seperated) by using the `--paths` flag. You can find the component paths in the `status.componentPaths` section of the respective Repository. If you want to unlock all the components of the respective Repository, then this flag should be omitted.

```
